06-building-a-program-start-to-finish/writing-tests/output/spec/InvertedIndex.spec.js

```diff
@@ -0,0 +1,211 @@
+    });
+
+    it('should check that the class has a searchIndex method', () => {
+      expect(typeof indexInstance.searchIndex).toBe('function');
+    });
+
+    it('should check that the class has a getSearchResults method', () => {
+      expect(typeof indexInstance.getSearchResults).toBe('function');
+    });
+
+    it('should check that the class has a getDocuments method', () => {
+      expect(typeof indexInstance.getDocuments).toBe('function');
+    });
+  });
+
+  describe('Create Index', () => {
+    it('should return mapped indices to words in a JSON file', () => {
+      const expectedResult =
+        { alice: [0],
+          falls: [0],
+          into: [0],
+          a: [0, 1, 2],
+          rabbit: [0],
+          hole: [0],
+          and: [0, 1, 2],
+          enters: [0],
+          world: [0],
+          full: [0],
+          of: [0, 1, 2],
+          imagination: [0],
+          in: [0],
+          wonderland: [0],
+          an: [1, 2],
+          unusual: [1, 2],
+          alliance: [1, 2],
+          man: [1, 2],
+          elf: [1, 2],
+          dwarf: [1, 2],
+          wizard: [1, 2],
+          hobbit: [1, 2],
+          seek: [1, 2],
+          to: [1, 2],
+          destroy: [1, 2],
+          powerful: [1, 2],
+          ring: [1, 2],
+          the: [1, 2],
+          lord: [1, 2],
+          rings: [1, 2],
+          fellowship: [1, 2] };
+      indexInstance.createIndex(books, 'books');
+      expect(indexInstance.filesIndexed.books.index)
+        .toEqual(expectedResult);
+    });
+    it('should return false for incorrect document structure', () => {
+      const term = { t1: 'Welcome home', text: 'This is really home' };
+      expect(indexInstance.createIndex(term, 'term')).toBeFalsy();
+    });
+    it('should return false for file with no content', () => {
+      const term = {};
+      expect(indexInstance.createIndex(term, 'term')).toBeFalsy();
+    });
+  });
+  describe('Search Index', () => {
+    it('should search through single files that are indexed', () => {
+      const requiredOutput = { alice: [0],
+        and: [0, 1, 2],
+        unusual: [1, 2],
+        imagination: [0] };
+      const searchTerm = indexInstance
+        .searchIndex('Alice, and her unusual imagination', 'books');
+      expect(Object.keys(searchTerm[0].indexes))
+        .toEqual(Object.keys(requiredOutput));
+      expect(searchTerm[0].indexes).toEqual(requiredOutput);
+    });
+    it('should return false for an empty String', () => {
+      const term = '';
+      expect(indexInstance.searchIndex(term, 'books'))
+      .toBeFalsy();
+    });
+    it('should return an empty object for an words not found', () => {
+      const term = 'Aeroplane';
+      const expectedOutput = indexInstance.searchIndex(term, 'books');
+      expect(expectedOutput[0].indexes).toEqual({ });
+    });
+  });
+  describe('Tokenize words', () => {
+    it('should strip out special characters from excerpt in documents', () => {
```

Comment:
Are you stripping out special characters from **_excerpt_** in documents? Please correct this.